tests/unit_tests/deployment/test_language_container_deployer.py

```diff
@@ -0,0 +1,122 @@
+#########################################################
+# To be migrated to the script-languages-container-tool #
+#########################################################
+from pathlib import Path, PurePosixPath
+from unittest.mock import create_autospec, MagicMock
+import pytest
+from pyexasol import ExaConnection
+from exasol_bucketfs_utils_python.bucketfs_location import BucketFSLocation
+from exasol_transformers_extension.deployment.language_container_deployer import (
+    LanguageContainerDeployer, LanguageActivationLevel)
+
+
+@pytest.fixture(scope='module')
+def container_file_name() -> str:
+    return 'container_xyz.tag.gz'
+
+
+@pytest.fixture(scope='module')
+def container_file_path(container_file_name) -> Path:
+    return Path(container_file_name)
+
+
+@pytest.fixture(scope='module')
+def language_alias() -> str:
+    return 'PYTHON3_TEST'
+
+
+@pytest.fixture(scope='module')
+def container_bfs_path(container_file_name) -> str:
+    return f'bfsdefault/default/container/{container_file_name[:-7]}'
+
+
+@pytest.fixture(scope='module')
+def mock_pyexasol_conn() -> ExaConnection:
+    return create_autospec(ExaConnection)
+
+
+@pytest.fixture(scope='module')
+def mock_bfs_location(container_bfs_path) -> BucketFSLocation:
+    mock_loc = create_autospec(BucketFSLocation)
+    mock_loc.generate_bucket_udf_path.return_value = PurePosixPath(f'/buckets/{container_bfs_path}')
+    return mock_loc
+
+
+@pytest.fixture
+def container_deployer(mock_pyexasol_conn, mock_bfs_location, language_alias) -> LanguageContainerDeployer:
+    deployer = LanguageContainerDeployer(pyexasol_connection=mock_pyexasol_conn,
+                                         language_alias=language_alias,
+                                         bucketfs_location=mock_bfs_location)
+
+    deployer.upload_container = MagicMock()
+    deployer.activate_container = MagicMock()
+    deployer.get_language_settings = MagicMock()
```

Comment:
Instead of making the function `get_language_settings` public and overwriting it with a mock, I would suggest to extract completely from the deployer class and injecting it into the constructor as a parameter with default to its actual implementation